### DISCUSSION TITLE
Decrypt encrypted database password in mirth.properties file

### DISCUSSION THREAD

**Discussion by abhausb2** [nextgenhealthcare/connect#6324] [Q&A] [answered]
Is there a way to decrypt the encrypted database password in the mirth.properties file. I can remove the encryption.properties = 1, and that removes the encryption from database password and stores the password as plain text. Is there a better way to decrypt the password without changing the encryption.properties.

I was also able to get the keys stored in the keystore, but not sure what should I do next with those keys.

I would really appreciate any insights on this.

Thanks!!

**Comment by jonbartels** [comment, marked as answer]
Take a look at https://github.com/nextgenhealthcare/connect/blob/master/server/src/com/mirth/connect/server/controllers/DefaultConfigurationController.java#L1339

See how the password is decrypted and stored in memory in the `databaseConfig` object?

You can get an instance of the ConfigurationController then call `getDatabaseSettings()` and get the username, password, etc.

**Reply by abhausb2** [reply]
Thanks a bunch @jonbartels!! surprised to see how easy this solution was to get the database credentials using the ConfigurationController
